Bluetooth shell minor fix and bond info

joerchan opened the pull request:
Add command to print all existing bonds. Useful for verifying that bonds
are actually cleared when debugging.

Fix whitelist clear (wl-clear) shell command argument count. This
command takes no arguments.

joerchan committed:
Bluetooth: shell: Fix whitelist clear shell command argument count

Fix whitelist clear (wl-clear) shell command argument count. This
command takes no arguments.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: shell: Add command to print all bonds available

Add command to print all existing bonds. Useful for verifying that bonds
are actually cleared when debugging.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

jhedberg commented on subsys/bluetooth/shell/bt.c:
```diff
@@ -1122,6 +1122,40 @@ static int cmd_bondable(const struct shell *shell, size_t argc, char *argv[])
+	}
+
+	shell_print(ctx_shell, "Remote Identity: %s, status: %s", addr,
+		    connected ? "Connected" : "Disconnected");
```
This isn't completely correct, since `bt_conn_lookup_addr_le()` may return unconnected objects as well, in case some piece of code is hanging onto their `bt_conn` reference even though the state of the connection has become disconnected.

joerchan replied:
True. Switched to only printing the conn object pointer instead. Gives an indication to the state, although not sure if it is needed.

jhedberg replied:
It seems a bit arbitrary to me. Don't we have a separate command to list connections? (if we don't perhaps we should add one?)

joerchan replied:
We don't have one for connections. I can add one.
I wanted to give the user an indications of which bonds where connected or not. So having a connections command to print similar info as this command would achieve that.

jhedberg commented on subsys/bluetooth/shell/bt.c:
```diff
@@ -1122,6 +1122,40 @@ static int cmd_bondable(const struct shell *shell, size_t argc, char *argv[])
+	int *bond_count = (int *)user_data;
+	char addr[BT_ADDR_LE_STR_LEN];
+	bool connected = false;
+	struct bt_conn * conn;
```
minor: no space between the `*` and the variable name

joerchan replied:
Fixed.

jhedberg commented on subsys/bluetooth/shell/bt.c:
```diff
@@ -1122,6 +1122,38 @@ static int cmd_bondable(const struct shell *shell, size_t argc, char *argv[])
 
+static void bond_info(const struct bt_bond_info *info, void *user_data)
+{
+	int *bond_count = (int *)user_data;
```
minor: remove the explicit `(int *)` typecast. The whole idea with void pointers is that they can be implicitly cast to any other pointer type.

joerchan committed:
Bluetooth: shell: Add command to print all connections available

Add command to print all existing connections. Useful for verifying that
connections are actually released when debugging.
More information can be retrieved with `bt info <addr>` using the
address printed by this command for each connection.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

jhedberg reviewed (approved):
Looks good! It'd be nice to see the state of the connection objects as well, perhaps make `state2str` in  `subsys/bluetooth/host/conn.c` a public function? (would be called `bt_conn_state_str()` then I suppose)

jhedberg commented:
> perhaps make state2str in subsys/bluetooth/host/conn.c a public function?

Actually, scratch that, since we don't expose the state publicly. You could instead just resort to internal header files since the shell is considered internal to the stack.
